File: 28/Python/main.py
#!/usr/bin/python

######################################################################
# Author: Matias Grioni
# Created: 7/15/15
#
# Sums the diagonals of a spiral that is size n x n. The way this
# spiral is formed can be seen on Problem 28 of Project Euler. Note
# the problem asks for a 1001 x 1001 spiral, which by my algorithm
# is a spiral that has 501 levels.
######################################################################

import time

def sumSpiral(dim):
    return sum([4*(4*n*n-7*n+4) for n in range(2, ((dim + 1) / 2) + 1)]) + 1

print("Sums the diagonals of a spiral of numbers of size n x n")
n = int(raw_input("Enter n > "))

start = time.time()
result = sumSpiral(n)
end = time.time()

print("Result: {}".format(result))
print("{} s".format(end - start))

File: 28/Haskell/main.hs
{-
 - Author: Matias Grioni
 - Created: 7/16/15
 -
 - Sums the diagonals in an n x n spiral as defined by problem 28 in
 - project euler.
 -}

module Main where

main = do
    putStrLn "Sum diagonals in an n x n spiral"
    putStrLn "Enter n > "
    input <- getLine
    let n = read input :: Int


    print $ 1 + sum [4*(4*n*n - 7*n + 4) | n <- [2..(n + 1) `div` 2]]
